snapcraft/_options.py

```diff
@@ -0,0 +1,41 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+class _Project:
+
+    @property
+    def use_geoip(self):
+        return self.__use_geoip
+
+    @use_geoip.setter
+    def use_geoip(self, setting):
+        self.__use_geoip = setting
+
+    def __init__(self):
+        self.__use_geoip = False
+
+
+_project = _Project()
```

Comment:
El miércoles, 6 de abril de 2016 14h'51:44 ART, Kyle Fazzari 
<notifications@github.com> escribió:
>> +
>> +class _Project:
>> +
>> +    @property
>> +    def use_geoip(self):
>> +        return self.__use_geoip
>> +
>> +    @use_geoip.setter
>> +    def use_geoip(self, setting):
>> +        self.__use_geoip = setting
>> +
>> +    def __init__(self):
>> +        self.__use_geoip = False
>> +
>> +
>> +_project = _Project()
> 
> Not a huge fan of introducing more global state. Would it be 
> dreadful to pass this around?

I don't see a way around this.Also this was supposed to be step one to 
encapsulate everything in `common` to this `_Project` class.

This looks pretty similar to how the logger package in python is 
implemented, except we'd be using a single element instead of N.

Also, if we pass this around I'd like to pass all of these in `common` 
around. It will be huge as there will always be more. Thoughts?


-- 
Enviado con Dekko desde mi dispositivo Ubuntu